test/controllers/api/v2/hosts_controller_test.rb

```diff
@@ -1230,4 +1230,36 @@ def setup
     assert_response :success
     assert_equal puppet_proxy['name'], JSON.parse(@response.body)['puppet_proxy']['name'], "Can't update host with puppet proxy #{puppet_proxy}"
   end
+
+  # This is a test of the base_controller functionality, but we need to use a real endpoint
+  # to test the API response metadata is returned correctly
+  describe 'should create correct subtotals' do
+    before do
+      @empty_org = Organization.find_by_name("Empty Organization")
+      as_admin do
+        FactoryBot.create_list(:host, 100, organization: @empty_org)
+      end
+    end
+
+    it 'with no search parameter' do
+      get :index
+      hosts_response = ActiveSupport::JSON.decode(@response.body)
+      host_count = Host.all.count
+
+      assert_response :success
+      assert_equal hosts_response["subtotal"], host_count
+      assert_equal hosts_response["total"], host_count
+      assert hosts_response["subtotal"] > hosts_response["per_page"]
+    end
+
+    it 'with search parameter' do
+      get :index, params: { search: "organization = \"#{@empty_org.name}\""}
+      hosts_response = ActiveSupport::JSON.decode(@response.body)
+
+      assert_response :success
+      assert_equal hosts_response["subtotal"], @empty_org.hosts.count
+      assert_equal hosts_response["total"], Host.all.count
```

Comment:
There is a hidden assumption here, that not all hosts are in `@empty_org` - perhaps assert that the subtotal and total are different?